pkg/rancher-desktop/main/extensions/extensions.ts

```diff
@@ -171,25 +178,84 @@ export class ExtensionImpl implements Extension {
 
     await Promise.all(paths.map(async(p) => {
       try {
-        await this.client.copyFile(this.id, p, binDir);
+        await this.client.copyFile(this.id, p, binDir, { namespace: this.extensionNamespace });
       } catch (ex: any) {
         throw new ExtensionErrorImpl(ExtensionErrorCode.FILE_NOT_FOUND, `Could not copy host binary ${ p }`, ex);
       }
     }));
   }
 
+  get containerName() {
+    const normalizedId = this.id.toLowerCase().replaceAll(/[^a-z0-9_-]/g, '_');
+
+    return `rd-extension-${ normalizedId }`;
+  }
+
+  /**
+   * Extra the Docker Compose files into a newly-created temporary directory.
+   * This is required because nerdctl doesn't seem to keep the definition around,
+   * @note The caller is expected to remove the output directory.
+   * @param composePath the value of metadata.vm.composefile
+   */
+  protected async extractComposeDefinition(composePath: string): Promise<string> {
+    const workDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-ext-install-'));
+    const composeDir = path.posix.dirname(path.posix.normalize(composePath));
+
+    await this.client.copyFile(
+      this.id,
+      composeDir === '.' ? '/' : `${ composeDir }/`,
+      workDir,
+      { namespace: this.extensionNamespace });
+
+    return workDir;
+  }
+
+  protected async installContainers(workDir: string, metadata: ExtensionMetadata): Promise<void> {
+    function isImage(input: any): input is { image: string } {
+      return typeof input?.image === 'string';
+    }
+
+    function isComposefile(input: any): input is { composefile: string } {
+      return typeof input?.composefile === 'string';
+    }
+
+    if (isImage(metadata.vm)) {
+      // eslint-disable-next-line no-template-curly-in-string -- literal ${DESKTOP_PLUGIN_IMAGE}
+      const imageID = metadata.vm.image === '${DESKTOP_PLUGIN_IMAGE}' ? this.id : metadata.vm.image;
+      const stdout = await this.client.run(imageID, {
+        namespace: this.extensionNamespace,
+        name:      this.containerName,
+        restart:   'always',
+      });
+
+      console.debug(`Running ${ this.id } container image ${ imageID }: ${ stdout.trim() }`);
+    } else if (isComposefile(metadata.vm)) {
+      const workDir = await this.extractComposeDefinition(metadata.vm.composefile);
+
+      try {
+        console.debug(`Running ${ this.id } compose up (workDir=${ workDir })`);
+        await this.client.composeUp(
+          workDir,
+          {
+            name:      this.containerName,
+            namespace: this.extensionNamespace,
+            env:       { DESKTOP_PLUGIN_IMAGE: this.id },
+          },
+        );
+      } finally {
+        // await fs.promises.rm(workDir, { recursive: true });
```

Comment:
Why is this comment here? If we want to leave it here, could you provide a quick explanation?

Reply:
My bad, this should be uncommented and the next line removed. (I had this for debugging.)